### PR TITLE
aws-sdk: add batchPropagationEnabled env variable configuration

### PR DESCRIPTION
**Pull request by wconti27**
### What does this PR do?
This PR adds the following env variables:
`DD_TRACE_AWS_SDK_BATCH_PROPAGATION_ENABLED` and `DD_TRACE_AWS_SDK_<SERVICE>_BATCH_PROPAGATION_ENABLED` . These env variables enable Datadog APM trace context injection for each message in batch send operations, as opposed to only injecting the first message in batches. The configuration is available for the following supported AWS SDK messaging services:
- SQS
- SNS
- Kinesis

### Motivation
Customer request

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

**Review by tlhunter — CHANGES_REQUESTED** [review]
This will need a test, plus it appears to be breaking existing tests?

File: packages/datadog-plugin-aws-sdk/src/base.js [hunk 1]
```diff
@@ -163,11 +163,34 @@ function normalizeConfig (config, serviceIdentifier) {
       break
   }
 
-  return Object.assign({}, config, specificConfig, {
+  const baseAWSBatchPropagationValue = process.env.DD_TRACE_AWS_SDK_BATCH_PROPAGATION_ENABLED
```
Comment (tlhunter) [inline packages/datadog-plugin-aws-sdk/src/base.js:166]: I think these should have correlating config values for when a user calls `tracer.init({})` manually.

File: packages/datadog-plugin-aws-sdk/src/base.js [hunk 2]
```diff
@@ -163,11 +163,34 @@ function normalizeConfig (config, serviceIdentifier) {
...
+  const baseAWSBatchPropagationEnabled = baseAWSBatchPropagationValue ? isTrue(baseAWSBatchPropagationValue) : false
+
+  // check if AWS batch propagation or AWS_[SERVICE] batch propagation is enabled via env variable
+  const serviceId = serviceIdentifier.toUpperCase()
```
Comment (tlhunter) [inline packages/datadog-plugin-aws-sdk/src/base.js:170]: Do the service identifiers consist entirely of a-z characters? Are there ever hyphens?

**Review by Qard — DISMISSED** [review]

File: packages/datadog-plugin-aws-sdk/src/base.js (hunk 2 above)
Comment (wconti27) [reply packages/datadog-plugin-aws-sdk/src/base.js:176]: I'm not 100% sure, but we already use this pattern within line 109 of this file and it hasn't caused any problems

**Comment by wconti27** [comment]
> This will need a test, plus it appears to be breaking existing tests?

I've simplified the configuration and added a test case.

File: packages/datadog-plugin-aws-sdk/src/base.js (hunk 1 above)
Comment (wconti27) [reply packages/datadog-plugin-aws-sdk/src/base.js]: I simplified this, they are now one and the same.

**Review by tlhunter — APPROVED** [review]

**Pull request merged**